### PR TITLE
utils: update curl to 8.9.1 from 8.5.0

### PR DESCRIPTION
**Pull request by compnerd**
Update the embedded curl version for Windows. Adjust the flags accordingly.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @etcwilde

**Comment by compnerd** [comment]
@swift-ci please test

**Review by etcwilde — COMMENTED** [review]
Please also update the build-script curl product with the new flags. Thanks.

**Comment by compnerd** [comment]
@swift-ci please test

**Review by etcwilde — APPROVED** [review]

**Comment by etcwilde** [comment]
@swift-ci Please Build Toolchain Linux Platform

**Comment by etcwilde** [comment]
So it looks like we're not even using the curl that we just built for Foundation while building Foundation.
```
-- Found CURL: /usr/lib/x86_64-linux-gnu/libcurl.so (found version "7.68.0")
```
(╯°□°）╯︵ ┻━┻ 

*sigh* Okay, this has no effect on Linux apparently. 🙄

File: utils/update_checkout/update-checkout-config.json [hunk 1]
```diff
@@ -160,7 +160,7 @@
                 "swift-sdk-generator": "main",
                 "wasi-libc": "wasi-sdk-20",
                 "wasmkit": "0.0.3",
-                "curl": "curl-8_5_0",
+                "curl": "curl-8_9_1",
```
Comment (shahmishal) [inline utils/update_checkout/update-checkout-config.json:163]: Please also update the next and rebranch.

Comment (compnerd) [reply utils/update_checkout/update-checkout-config.json:163]: https://github.com/swiftlang/swift/pull/75868

**Pull request merged**